File: core/src/ca/on/hojat/snakegame/dragrace/objects/InfiniteScrollBg.java
package ca.on.hojat.snakegame.dragrace.objects;

import ca.on.hojat.snakegame.dragrace.Assets;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.Actor;
import com.badlogic.gdx.scenes.scene2d.actions.MoveToAction;

import static com.badlogic.gdx.scenes.scene2d.actions.Actions.*;

public class InfiniteScrollBg extends Actor {

	MoveToAction moveAction;

	public InfiniteScrollBg(float width, float height) {
		setWidth(width);
		setHeight(height);
		setPosition(0, height);

		moveAction = new MoveToAction();
		moveAction.setPosition(0, 0);
		moveAction.setDuration(1.75f);

		addAction(forever(sequence(moveAction, moveTo(0, height))));

	}

	@Override
	public void draw(Batch batch, float parentAlpha) {
		super.draw(batch, parentAlpha);
		batch.draw(Assets.calle, getX(), getY() - getHeight(), getWidth(),
				getHeight() * 2);
	}

	public void setSpeed() {
		moveAction.setDuration(.3f);
	}

	public void stopSpeed() {
		moveAction.setDuration(1.75f);
	}

}

File: core/src/ca/on/hojat/snakegame/dragrace/objects/InfiniteScrollBg.kt
package ca.on.hojat.snakegame.dragrace.objects

import ca.on.hojat.snakegame.dragrace.Assets
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.actions.Actions
import com.badlogic.gdx.scenes.scene2d.actions.MoveToAction

/**
 * Our infinitely scrolling 2D background is an [Actor]
 */
class InfiniteScrollBg(
    private val backGroundWidth: Float,
    private val backGroundHeight: Float
) : Actor() {

    private val moveAction = MoveToAction()

    init {
        setPosition(0f, backGroundHeight)
        moveAction.setPosition(0f, 0f)
        moveAction.duration = 1.75f
        addAction(
            Actions.forever(
                Actions.sequence(
                    moveAction, Actions.moveTo(0f, backGroundHeight)
                )
            )
        )
    }

    override fun draw(batch: Batch?, parentAlpha: Float) {
        super.draw(batch, parentAlpha)
        batch?.draw(
            Assets.calle,
            x,
            y - backGroundHeight,
            backGroundWidth,
            backGroundHeight * 2
        )
    }

    fun setSpeed() {
        moveAction.duration = .3f
    }

    fun stopSpeed() {
        moveAction.duration = 1.75f
    }
}